File: crossecore-java/src/java_/InfixExpression.java
package java_;
public interface InfixExpression 
 extends Expression
{
	
	public static com.crossecore.ocl.QuickSet<InfixExpression> allInstances_ = new com.crossecore.ocl.QuickSet<InfixExpression>(InfixExpression.class);
	
		
	public static com.crossecore.ocl.QuickSet<InfixExpression> allInstances(){
		
		com.crossecore.ocl.QuickSet<InfixExpression> result = new com.crossecore.ocl.QuickSet<InfixExpression>(InfixExpression.class);
		result.addAll(InfixExpression.allInstances_);
		
		
		return result;
	}
	
	//public static com.crossecore.ocl.QuickSet<InfixExpression> allInstances = new com.crossecore.ocl.QuickSet<InfixExpression>(InfixExpression.class);
	
	InfixExpressionKind getOperator();
	void setOperator(InfixExpressionKind value);
	
	Expression getRightOperand();
	void setRightOperand(Expression value);
	
	Expression getLeftOperand();
	void setLeftOperand(Expression value);
	com.crossecore.ocl.OrderedSet<Expression> getExtendedOperands();
	
}

File: crossecore-typescript/java_/InfixExpression.ts
import {OrderedSet} from "ecore/OrderedSet";
import {Set} from "ecore/Set";
import {Bag} from "ecore/Bag";
import {InfixExpressionKind} from "./InfixExpressionKind";
import {Expression} from "./Expression";

export interface InfixExpression
extends Expression

{
	operator:InfixExpressionKind;
	
	rightOperand:Expression;
	leftOperand:Expression;
	extendedOperands: OrderedSet<Expression>;
	

}

